ethicalads.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"net/http"
+	"net/url"
+)
+
+type EthicalAdsAd struct {
+	Ad
+	Pixel []string
+}
+
+type EthicalAdsResponse struct {
+	Id      string
+	Body    string
+	Image   string
+	Link    string
+	ViewUrl string `json:"view_url"`
+	Nonce   string
+}
+
+var hystrixEa = "EthicalAds"
+
+var fetchEthicalAds = func(r *http.Request) (*EthicalAdsAd, error) {
+	params := url.Values{}
+	params.Add("callback", "ethicalads")
+	params.Add("format", "json")
+	params.Add("publisher", "dailydev")
+	params.Add("div_ids", "sample-ad")
+	params.Add("ad_types", "image-v1")
+	params.Add("user_ip", getIpAddress(r))
+	params.Add("user_ua", r.UserAgent())
+	var res EthicalAdsResponse
+	req, _ := http.NewRequest("GET", "https://server.ethicalads.io/api/v1/decision/?"+params.Encode(), nil)
```

Comment:
I think for doing a server-to-server call we should probably use a POST instead of a GET. We don't currently support the `user_ip` and `user_ua` in the GET requests, because those generally come from the client itself. Otherwise this looks great 👍

Reply:
Thanks for the comment :)
So how should I provide the parameters for the POST request?
Raw body as JSON?

Reply:
Yea, it's just JSON in the body.

The other thing to consider is to add [auth](https://ethical-ad-server.readthedocs.io/en/latest/user-guide/api.html#authentication) to the request. We have an option to only allow authenticated requests from a publisher, and that will ensure that only the request are coming from your account. You'd want to keep that in an environment variable or something, not in the code. We can hold off on that for now, but we do recommend it for this usage.